js/service.js

```diff
@@ -50,7 +50,8 @@ function toQueryString(data) {
  */
 function getDataFromURL(route, params = {}) {
   let request = new XMLHttpRequest();
-  let combinedParams = Object.assign(defaultParams, params);
+  let clonedDefaultParams = Object.assign({}, defaultParams); // so we don't mutate defaultParams itself in the next line
+  let combinedParams = Object.assign(clonedDefaultParams, params);
```

Comment:
No need to split this over three lines, Object.assign thankfully takes "as many objects as necessary":
```
let combinedParams = Object.assign({}, defaultParams, params);
```